airbyte_ui_rock/rockcraft.yaml

```diff
@@ -0,0 +1,75 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+name: airbyte-ui
+summary: Airbyte UI rock
+description: Airbyte UI OCI image for the Airbyte UI charm
+version: "1.0"
+base: ubuntu@22.04
+build-base: ubuntu@22.04
+license: Apache-2.0
+platforms:
+  amd64:
+
+services:
+  airbyte-webapp:
+    override: replace
+    summary: "airbyte-webapp service"
+    startup: disabled
+    command: "/usr/bin/pnpm -C airbyte-webapp start"
+    environment:
+      PORT: "8080"
+
+parts:
+  patches:
+    plugin: dump
+    source: ./patches
+    organize:
+      vite-config-http.patch: patches/vite-config-http.patch
+    stage:
+      - patches/vite-config-http.patch
+    prime:
+      - "-*"
+
+  airbyte-webapp:
+    after: [patches]
+    plugin: dump
+    source: https://github.com/airbytehq/airbyte-platform.git # yamllint disable-line
+    source-type: git
+    source-tag: v0.63.8
+    build-packages:
+      - jq
+      - curl
+      - nodejs
+      - npm
+      - coreutils
+      - bash
+    override-build: |
+      git apply ${CRAFT_STAGE}/patches/*.patch
+
+      curl https://raw.githubusercontent.com/creationix/nvm/v0.40.1/install.sh | bash
+      source ~/.bashrc
+      cd airbyte-webapp
+      nvm install
```

Comment:
This will install the latest LTS version of nodejs. If your application depends on certain version of nodejs, or there is a change it may break when the next LTS version is rolled out, then better to pin the version here.